lib/matplotlib/tight_layout.py

```diff
@@ -1,3 +1,16 @@
 from matplotlib._tight_layout import *  # noqa: F401, F403
 from matplotlib import _api
 _api.warn_deprecated("3.6", name=__name__, obj_type="module")
+
+
+# To be removed with the module
```

Comment:
Should this get its own deprecation message like: 
```
@_api.deprecated("3.6", alternative="figure.canvas.get_renderer()")
```

Reply:
I think this not necessary. The whole module was deprecated including all contained functions; and you get the warning at module level.

The function is only listed here and not in the renamed private copy because we do not need it internally. We could have left it in there, but then we'd need to remember deleting it when the expiration runs out. That's an internal detail and not of concern for our users.

Reply:
The specific depreciation message on the function would tell the user what to use instead.  Is there an easy/obvious way to find that otherwise?

Reply:
Now it will warn on import, but not on execution (I think). I can double deprecate.